docs/source/user-guide/sql/dialect.md

```diff
@@ -0,0 +1,38 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# SQL Dialect
```

Comment:
it would be good to say this section pertains the SQL frontend:

- sql parser and sql-to-rel
- dataframe API
- analyzer and the type coercions done at this stage
- function semantics of functions bundled with datafusion

see also https://github.com/apache/datafusion/issues/13704#issuecomment-2531210662

it's also worth noting that we are not going to align with PostgreSQL's dialect fully

- eg our type system is different (inherited from arrow)
- we provide extended syntax for certain operations (like CREATE TABLE)
- (...)

Reply:
Reworded and expanded in b674a42a25a29935492f3fc58a1fbcd14c954e22